lib/Test/Nginx/Socket.pm

```diff
@@ -1101,6 +1101,47 @@ sub check_error_log ($$$$) {
         }
     }
 
+    if (defined $block->access_log) {
+        my $pats = $block->access_log;
+
+        if (!ref $pats) {
+            chomp $pats;
+            my @lines = split /\n+/, $pats;
+            $pats = \@lines;
+
+        } elsif (ref $pats eq 'Regexp') {
+            $pats = [$pats];
+
+        } else {
+            my @clone = @$pats;
+            $pats = \@clone;
+        }
+
+        $lines ||= access_log_data();
```

Comment:
You are sharing the `$lines` variable with error logs here? It looks very bad since you are essentially mixing error logs and access logs here.